src/playlist-maker.js

```diff
@@ -133,217 +122,336 @@ const factory = (player, initialList, initialIndex = 0) => {
+   * @fires playlistsorted
+   *
+   * @param {Function} compare
+   *        A comparator function as per the native Array method.
+   */
+  playlist.sort = (compare) => {
+
+    // Bail if the array is empty.
+    if (!list.length) {
+      return;
+    }
+
+    list.sort(compare);
 
     /**
-     * Plays the previous item in the playlist.
+     * Triggered after the playlist is sorted internally.
      *
-     * @return {Object|undefined}
-     *         Returns undefined and has no side effects if on first item.
+     * @event playlistsorted
+     * @type {Object}
      */
-    previous() {
+    player.trigger('playlistsorted');
+    return list;
+  };
 
-      // Make sure we don't go past the start of the playlist.
-      const index = Math.max(playlist.currentIndex_ - 1, 0);
+  /**
+   * Reverses the playlist array.
+   *
+   * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reverse}
+   * @fires playlistsorted
+   */
+  playlist.reverse = () => {
 
-      if (index !== playlist.currentIndex_) {
-        return list[playlist.currentItem(index)];
-      }
-    },
+    // Bail if the array is empty.
+    if (!list.length) {
+      return;
+    }
+
+    list.reverse();
 
     /**
-     * Sets up auto-advance on the playlist.
+     * Triggered after the playlist is sorted internally.
      *
-     * @param {number} delay
-     *        The number of seconds to wait before each auto-advance.
+     * @event playlistsorted
+     * @type {Object}
      */
-    autoadvance(delay) {
-      playlist.autoadvance_.delay = delay;
-      autoadvance.setup(playlist.player_, delay);
-    },
+    player.trigger('playlistsorted');
+    return list;
+  };
+
+  /**
+   * Shuffle the contents of the list randomly.
+   *
+   * @see {@link https://github.com/lodash/lodash/blob/40e096b6d5291a025e365a0f4c010d9a0efb9a69/shuffle.js}
+   * @fires playlistsorted
+   */
+  playlist.shuffle = () => {
+    let index = -1;
+    const length = list.length;
+
+    // Bail if the array is empty.
+    if (!length) {
+      return;
+    }
+
+    const lastIndex = length - 1;
+
+    while (++index < length) {
+      const rand = index + Math.floor(Math.random() * (lastIndex - index + 1));
+      const value = list[rand];
+
+      list[rand] = list[index];
+      list[index] = value;
+    }
 
     /**
-     * Sets `repeat` option, which makes the "next" video of the last video in the
-     * playlist be the first video in the playlist.
+     * Triggered after the playlist is sorted internally.
      *
-     * @param {boolean=} val
-     *        The value to set repeat to
-     *
-     * @return {boolean}
-     *         The current value of repeat
+     * @event playlistsorted
+     * @type {Object}
      */
-    repeat(val) {
-      if (val !== undefined) {
-        if (typeof val !== 'boolean') {
-          videojs.log.error('Invalid value for repeat', val);
```

Comment:
Removed this strict type check as it seemed like overkill.

Reply:
it's like that due to a compromise: https://github.com/brightcove/videojs-playlist/pull/71#discussion_r105244134

Reply:
That may be, but it makes no sense to me. It's inconsistent with the rest of the API and I don't see a reason why doing `repeat('true')` or `repeat(1)` shouldn't work when that kind of strictness is usually not enforced in any of our APIs...

Reply:
I'll roll it back, though. Not looking to get into a debate about API design choices.

Reply:
Rolled back. Log message will still appear.